yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -290,8 +290,18 @@ private[yarn] class YarnAllocator(
       location: String,
       containersToUse: ArrayBuffer[Container],
       remaining: ArrayBuffer[Container]): Unit = {
+    // SPARK-6050: certain Yarn configurations return a resource configuration that doesn't match
+    // the request; for example, capacity scheduler + DefaultResourceCalculator. Allow users in
+    // those situations to disable resource matching.
+    val matchingResource =
+        if (sparkConf.getBoolean("spark.yarn.container.matchAnyResource", false)) {
```

Comment:
re: memory, can yarn really allocate a container with less resources than you asked for?

The `resource` in this class is pretty much static during the Spark app's lifetime.